server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -808,7 +809,7 @@ protected String createLocalStoragePoolName(Host host, StoragePoolInfo storagePo
     @Override
     public PrimaryDataStoreInfo createPool(CreateStoragePoolCmd cmd) throws ResourceInUseException, IllegalArgumentException, UnknownHostException, ResourceUnavailableException {
         String providerName = cmd.getStorageProviderName();
-        Map<String,String> uriParams = extractUriParamsAsMap(cmd.getUrl());
+        Map<String,String> uriParams = extractUriParamsAsMap(cmd.getUrl(), cmd.isManaged());
```

Comment:
@shwstppr, sorry github doesn't allow you to comment on lines without any changes. On lines 825 and 897  there will be null pointer if the uriParams is empty.

```
 825       ScopeType scopeType = uriParams.get("scheme").toString().equals("file") ? ScopeType.HOST : ScopeType.CLUSTER;

 897       if (params.get("scheme").toString().equals("file")) 
```